pkg/virt-controller/watch/vm.go

```diff
@@ -222,6 +225,10 @@ func (c *VMController) execute(key string) error {
 		return err
 	}
 
+	if err := handlePVCMisuseInVM(c.pvcInformer, c.recorder, vm); err != nil {
```

Comment:
This needs to go into the `startVMI()` function.

Where this check occurs now would actually prevent us from stopping an already running VM if someone attempted to add a new volume to the VMI template and restart the VM.